### ISSUE TITLE
Work plan

### ISSUE THREAD

**Issue by Fuuzetsu** [hstorrent/hstorrent#1]
So there's quite a few of us now so perhaps we should code some stuff.  I think implementing 
https://wiki.theory.org/BitTorrentSpecification and then extending it with all the BEPs is the natural way.

Implementation then verification and optimisation then extension seems like a good way to go.

If someone is interested in implementing a specific part then please speak up. From the list below, the first two are fairly trivial as libraries exist.
- Bencoding
- Metainfo File Structure
- Tracker HTTP/HTTPS Protocol
- Tracker 'scrape' Convention
- Peer wire protocol (TCP)

Basically start hacking on some parts, I don't have a grand plan on the order of implementation so pick out some work and go at it! Of course if someone does have some kind of a work plan they want to impose then feel free to share it.

As usual, #haskell-bittorrent is the channel to idle in.

cc @hstorrent/contributors

**Comment by Fuuzetsu** [comment]
FTR going off IRC, just so we don't duplicate work:
- @osa1 is implementing the torrent type + bencoding
- @qleguennec is working on Tracker HTTP/HTTPS Protocol
- I'm working on Peer wire protocol (TCP)

**Comment by Fuuzetsu** [comment]
@qleguennec any progress?

**Comment by qleguennec** [comment]
If anything goes as i planned, i'll be able to send a pull request by the
end of the week-end.

2014-09-04 23:52 GMT+02:00 Mateusz Kowalczyk notifications@github.com:

> @qleguennec https://github.com/qleguennec any progress?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/hstorrent/hstorrent/issues/1#issuecomment-54550750.

## 

Quentin Le Guennec

**Comment by qleguennec** [comment]
Well, everything didn't exactly went as i planned. Turns out i'm very busy. I pushed Request.hs on my [fork](https://github.com/qleguennec/hstorrent/blob/master/src/Network/HsTorrent/Request.hs)
I don't know if this is blocking anyone, so if it is, poke me and i'll try to write specific parts that are most needed. Or, if anyone wants to hack my part, do it.

**Comment by Fuuzetsu** [comment]
Tracker is pretty important part because it enables us to test other everything else against real data. Maybe I can find time for it to help you out.
